holoviews/operation/datashader.py

```diff
@@ -319,19 +318,109 @@ def _process(self, element, key=None):
             return NdOverlay(layers, kdims=[data.get_dimension(column)])
 
 
 
+class regrid(resample_operation):
+    """
+    regrid allows resampling a HoloViews Image type using specified
+    up- and downsampling functions defined using the aggregator and
+    interpolation parameters respectively. By default upsampling is
+    disabled to avoid unnecessarily upscaling an image that has to be
+    sent to the browser. Also disables expanding the image beyond its
+    original bounds avoiding unneccessarily padding the output array
+    with nan values.
+    """
+
+    aggregator = param.ObjectSelector(default='mean',
+        objects=['first', 'last', 'mean', 'mode', 'std', 'var', 'min', 'max'], doc="""
+        Aggregation method.
+        """)
+
+    expand = param.Boolean(default=False, doc="""
+       Whether the x_range and y_range should be allowed to expand
+       beyond the extent of the data.  Setting this value to True is
+       useful for the case where you want to ensure a certain size of
+       output grid, e.g. if you are doing masking or other arithmetic
+       on the grids.  A value of False ensures that the grid is only
+       just as large as it needs to be to contain the data, which will
+       be faster and use less memory if the resulting aggregate is
+       being overlaid on a much larger background.""")
+
+    interpolation = param.ObjectSelector(default='nearest',
+        objects=['linear', 'nearest'], doc="""
+        Interpolation method""")
+
+    upsample = param.Boolean(default=False, doc="""
+        Whether to allow upsampling if the source array is smaller than
+        the requested array.""")
+
+    def _process(self, element, key=None):
+        if ds_version <= '0.5.0':
+            raise RuntimeError('regrid operation requires datashader>=0.6.0')
+
+        x, y = element.kdims
+        (x_range, y_range), _, (width, height) = self._get_sampling(element, x, y)
+
+        coords = tuple(element.dimension_values(d, expanded=False)
+                       for d in [x, y])
+        if element.interface is XArrayInterface and len(element.vdims) == 1:
+            xarr = element.data[element.vdims[0].name]
+        else:
+            coord_dict = {x.name: coords[0], y.name: coords[1]}
+            dims = [y.name, x.name]
+            arrays = []
+            for vd in element.vdims:
+                arrays.append(element.dimension_values(vd, flat=False))
+            if len(arrays) > 1:
+                array = np.dstack(arrays).transpose([2, 0, 1])
+                coords['layer'] = range(len(arrays))
+                dims = ['layer'] + dims
+            else:
+                array = arrays[0]
+            xarr = xr.DataArray(array, coords=coord_dict, dims=dims)
+
+        # Disable upsampling if requested
+        (xstart, xend), (ystart, yend) = (x_range, y_range)
+        xspan, yspan = (xend-xstart), (yend-ystart)
+        if not self.p.upsample and self.p.target is None:
```

Comment:
In that case the docstring needs improvement:

"Whether to allow upsampling if the source array is smaller than the requested array."

What is the behavior if ``upsample=False``? Padding?